### ISSUE TITLE
Android: Multiple notifcations for devices with same characteristic ids

### ISSUE THREAD

**Issue by xabre** [xabre/xamarin-bluetooth-le#107]
While thinking about the implementation of #90 I noticed the following case:

- Suppose we are connected to two devices of the same make at the same time (same services and characteristic ids) . 
- We start notifcations for **device1**(d1) -> s -> characteristic(c) and subscribe with **handler1** (h1)
- We start notifcations for **device2**(d2) -> s -> characteristic(c) and subscribe with **handler2** (h2)
- Basically we subscribe for notifcations on the same characteristic on both devices.
- We receive a notification only from **d1** -> c and both are handlers (h1 and h2) are invoked because we only filter by characteristic id an not device id when raising value changed events.

@smstuebe what do you think?

**Comment by smstuebe** [comment]
let us have a look at this in office :) Currently preparing some talks, xamarin dev days and some other community stuff ^^

**Comment by xabre** [comment]
#144 should solve this
